lib/compilers/dex2oat.ts

```diff
@@ -211,14 +211,17 @@ export class Dex2OatCompiler extends BaseCompiler {
             'bootjars/apache-xml.jar',
         ];
 
+        const versionFile = this.artArtifactDir + '/snapshot-creation-build-number.txt';
+        const buildId = parseInt(fs.readFileSync(versionFile, {encoding: 'utf8'}));
+
+        const artVersion3310 = 9106705;
         const dex2oatOptions = [
             '--android-root=include',
             '--generate-debug-info',
             '--dex-location=/system/framework/classes.dex',
             `--dex-file=${d8DirPath}/${dexFile}`,
             '--copy-dex-files=always',
-            '--runtime-arg',
-            '-Xgc:CMC',
+            ...(buildId > artVersion3310 ? ['--force-allow-oj-inlines', '--runtime-arg', '-Xgc:CMC'] : []),
```

Comment:
Could you please try removing '--force-allow-oj-inlines'? I think it's only need when generating the boot image?

Reply:
I tried testing with --force-allow-oj-inlines removed, but I'm not sure what case I can use to test correctness. I did take a look at the log output, and the "do not distribute" log that I was expecting is no longer there—would there be any harm in leaving the flag in?

Reply:
It's correct as long as the code compiles. I don't think we expect the "do not distribute" log. What can't be distributed is the boot image generated with --force-allow-oj-inlines. The app is fine.

Reply:
Got it, I've removed the flag. Thanks!